spi-proposal-no-transaction.md

```diff
@@ -0,0 +1,256 @@
+12: csPin                 (0-127) The chip select pin number
+13: END_SYSEX
+```
+
+#### bitOrder
+```
+LSBFIRST = 0
+MSBFIRST = 1 (default)
+```
+
+#### dataMode
+
+| mode    | clock polarity (CPOL) | clock phase (CPHA) |
+| --------|-----------------------|--------------------|
+| 0       | 0                     | 0                  |
+| 1       | 0                     | 1                  |
+| 2       | 1                     | 0                  |
+| 3       | 1                     | 1                  |
+
+#### maxSpeed
+
+The maximum speed of communication with the SPI device. For a SPI device rated
+up to 5 MHz, use 5000000.
+
+*For platforms that use a clock divider instead of a speed, pass the clock
+divider value instead.* 
+
+#### wordSize
+
+The size of a `word` in bits. Typically 8-bits (default). 0 = DEFAULT = 8-bits,
+1 = 1 bit, 2 = 2 bits, etc (limit is TBD).
+
+#### csPinOptions / csPin
+
+To enable automatic control of the csPin, set bit 0 (`CS_PIN_CONTROL`) to 1
+(enable). To manually control the csPin state with `DIGITAL_MESSAGE` commands,
+set `CS_PIN_CONTROL` to 0 (disable).
+
+The default behavior is to pull the CS pin LOW at the beginning of a transaction
+and toggle it at the end of the transaction. Setting bit 1 (`CS_ACTIVE_STATE`)
+will pull the CS pin HIGH at the start of the transaction.
+
+If `CS_PIN_CONTROL` is set to 0 (disable), the csPin parameter is ignored. Set
+the value to zero. The user will then need to manually control the csPin state
+using `DIGITAL_MESSAGE` commands.
+
+### SPI_TRANSFER
+
+Full-duplex write/read transfer. This is the normal SPI transfer mode, a word
+must be written for every word read. Reply is sent via `SPI_REPLY`.
+
+The user can continue sending as many transfer command as necessary, even
+alternating between `SPI_TRANSFER`, `SPI_WRITE` and `SPI_READ` within a single
+SPI transaction.
+
+`requestId` is used in the request messages `SPI_TRANSFER`, `SPI_WRITE` and
+`SPI_READ` and in the reply message `SPI_REPLY`. Its purpose is to ensure that
+the SPI_REPLY message matches the request. For each request message, increment
+a single 7-bit requestId value, rolling it over to 0 when > 127.
+
+`csPinChange` is used to control the csPin at the end of the transfer. By
+default the csPin will be deselected (the csPin will be toggled). However, to
+prevent deselection to enable back-to-back transfers for example, set
+`csPinChange` to 0 and the pin state won't be affected.
+
+If `CS_PIN_CONTROL` is enabled, then the csPin active state will be set when
+the `SPI_TRANSFER` command is received. It will only be deselected (toggled) at
+the end of the transfer if `csPinChange` is set to 1.
+
+```
+0:  START_SYSEX
+1:  SPI_DATA              (0x68)
+2:  SPI_TRANSFER          (0x02)
+3:  deviceId | channel    (bits 3-6: deviceId, bits 0-2: channel)
+4:  requestId             (0-127) // increment for each call
+5:  csPinChange           (0 = don't deselect csPin
```

Comment:
This is borrowed from the Linux SPI api and has the same behavior.

Reply:
I'd be tempted to pack this bit into the deviceId/channel byte, but that would lose 4 channels or 8 device Ids.